RangeReaderProgramTest/RangeReaderTest.cs

```diff
@@ -0,0 +1,43 @@
+using NSubstitute;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NSubstitute.ReceivedExtensions;
+
+namespace RangeReader
+{
+
+  public class RangeReaderTest
+  {
+    [Test]
+    [TestCase(3, 10, 4, new[] { 3, 3, 5, 4, 10, 11, 12 })]
```

Comment:
isn't the expected count supposed to be 5 - 3,3,5,4,10 ? i.e. range 3-10 +. count 5?